ballerina/googlesheets_client.bal

```diff
@@ -240,12 +240,9 @@ public isolated client class GoogleSheetsClient {
             return <persist:Error>error(rows.message());
         }
         if rows.length() == 0 {
-            if key is map<anydata> {
-                return <persist:NotFoundError>error(string `Not found: ${self.generateKeyArrayString(self.keyFields, key)}`);
-            } else {
-                return <persist:NotFoundError>error(string `Not found: ${key.toString()}`);
-            }
-        } else if rows.length() > 1 {
+            return persist:getNotFoundError(self.entityName, key);
+        }
+        else if rows.length() > 1 {
```

Comment:
```suggestion
        } else if rows.length() > 1 {
```

Reply:
Fixed in [6af274a](https://github.com/ballerina-platform/module-ballerinax-persist.googlesheets/pull/15/commits/6af274ac3b3c8e6595db3232d142746015cd4ef0)